### PR TITLE
GAWB-3625 Validate method configurations properly

### PR DESCRIPTION
**Pull request by breilly2**
(Do-over of #1322 which I can't reopen after force-pushing the branch)

Friends with broadinstitute/rawls#945

Updated in line with some backend changes that correctly check the validity of method config expressions when inputs on the method are optional. Upshot of this is that the endpoint returns more info about precisely why your conf is wrong, so we handle that here.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs [hunk 1]
```diff
@@ -361,15 +367,46 @@
...
+   (fn [{:keys [state this]} unvalidated-mc]
+     (endpoints/call-ajax-orch
+      {:endpoint endpoints/get-inputs-outputs
+       :payload (:methodRepoMethod unvalidated-mc)
+       :headers ajax/content-type=json
+       :on-done (fn [{:keys [success? get-parsed-response]}]
+                  (if success?
+                    (this :-load-validated-method-config (get-parsed-response))
+                    (let [fake-inputs-outputs (fn [data]
+                                                (let [method-config (:methodConfiguration data)]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs]: Just have this function take `unvalidated-mc` so you don't need to unpack it again

**Review by MattPutnam — APPROVED** [review]
👍 seems reasonable

**Comment by breilly2** [comment]
Ignore automatic test run because these UI changes need corresponding rawls changes. This UI test run succeeded: https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/7636/

**Review by ansingh7115 — APPROVED** [review]
Looks good! Review comments were addressed on previous PR

**Comment by helgridly** [comment]
Bit biased obviously, but seems legit :+1:

**Pull request merged**